### PR TITLE
IAC-796 Upgrade VCD archetype to support Angular 15

### PR DESCRIPTION
**Pull request by vganevjb**
### Description
Upgrade VCD archetype from Angular 8 to Angular 15 and the corresponding dependencies.

### Checklist
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added relevant error handling and logging messages to help troubleshooting
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have updated CHANGELOG.md with a short summary of the changes introduced
- [x] I have tested against live environment, if applicable
- [ ] I have added relevant usage information (As-built)
- [ ] Any structure and/or content vRA-NG improvements are synchronized with vra-ng and ts-vra-ng archetypes
- [ ] Every new or updated Installer property is documented in docs/archive/doc/markdown/use-bundle-installer.md
- [ ] Dependencies in pom.xml are up-to-date
- [x] My changes have been rebased and squashed to the minimal number of relevant commits
- [x] My changes have a descriptive commit message with a short title, including a `Fixed #XXX -` or `Closed #XXX -` prefix to auto-close the issue

### CODE REVIEW

**Comment by VenelinBakalov** [comment]
Hi, @vganevjb . I see you updated CHANGELOG.md which is great. Could you updated Release.md as well?
Here is a sample: https://github.com/vmware/build-tools-for-vmware-aria/pull/190/files#diff-2758ec22240769ca24b53db71e3dc4dc4a7990dabed3bc38bf2ec4456d19f3a1

**Review by n-vodenicharski — CHANGES_REQUESTED** [review]
Please investigate from which version of vCD, Angular v15 can be used instead of v8.
The approach will be to move the old code to new archetype for v8 and update the existing archetype for v15 .

File: docs/versions/latest/Release.md
Comment (ivo-kotev) [inline docs/versions/latest/Release.md:103]: @vganevjb  What is the minimum VCD version to support Angular 15?

**Comment by ivo-kotev** [comment]
> Please investigate from which version of vCD, Angular v15 can be used instead of v8. The approach will be to move the old code to new archetype for v8 and update the existing archetype for v15 .

Let's see since which VCD version angular 15 is supported. If it's supported for the version that are still under support (>= VCD 10.4), i guess we can change this archetype as it will work well for new deployments. Changing the archetype affects only the new projects, so old existing projects will still work without any changes.

**Comment by ivo-kotev** [comment]
I just had another conversation about this. Based on that, i think what @n-vodenicharski  proposed would make more sense. We will need to have Angular 2-9 projects available for all VCD instances running on 10.4 (EOL mid 2024) and we can switch to Angular 15 for all VCD projects >= 10.5.

File: docs/versions/latest/Release.md
Comment (vganevjb) [reply docs/versions/latest/Release.md]: I successfully tested on 10.4.0, 10.4.1, 10.4.2 and 10.5.0

**Comment by vganevjb** [comment]
@n-vodenicharski @ivo-kotev I added a new archetype vcd-ng-angular8-archetype for the older VCD versions that can be bootstrapped by:
`mvn archetype:generate \
    -DinteractiveMode=false \
    -DarchetypeGroupId=com.vmware.pscoe.vcd.archetypes \
    -DarchetypeArtifactId=package-vcd-ng-angular8-archetype \
    -DgroupId=org.example \
    -DartifactId=sample \
    -DlicenseUrl= \
    -DlicenseHeader= \
    -DlicenseTechnicalPreview=false
`

**Review by ivo-kotev — APPROVED** [review]

**Review by n-vodenicharski — APPROVED** [review]

**Comment by VenelinBakalov** [comment]
@vganevjb could you check, there are some linter issues blocking the merge

**Comment by VenelinBakalov** [comment]
Hmm seems the linter is still failing and the branch is out of date with the main. maybe we can organise a chat next week with the approvers so we can sync when to push the needed updates, immediately approve after that and merge so we can close this topic
@vganevjb @n-vodenicharski @ivo-kotev

**Comment by VenelinBakalov** [comment]
Hi team, @vganevjb @ivo-kotev @n-vodenicharski do we still plan to merge this feature?

**Comment by VenelinBakalov** [comment]
@vganevjb @n-vodenicharski @ivo-kotev team, are there any open actions on this pr?
@vganevjb I see we have some merge conflicts

**Comment by VenelinBakalov** [comment]
@n-vodenicharski @ivo-kotev please double check, i think this was already approved before and we should be ok with merging it

**Review by ivo-kotev — APPROVED** [review]

**Pull request merged**
